client/src/extension.ts

```diff
@@ -60,59 +60,36 @@ export function activate(context: vscode.ExtensionContext) {
 
           // brew install location mac
           // intel
-          "/usr/local/bin/nu",
+          "/usr/local/bin",
           // arm
-          "/opt/homebrew/bin/nu",
+          "/opt/homebrew/bin",
 
           // native package manager install location
-          "/usr/bin/nu",
+          // standard location should be in `PATH` env var
+          //"/usr/bin/nu",
         ];
 
-        let found_nushell_path = "";
-        const home = os.homedir();
-
-        for (const cur_val of pathsToCheck) {
-          // console.log("Inspecting location: " + cur_val);
-          let constructed_file = "";
-          if (cur_val.startsWith("~/scoop")) {
-            // console.log("Found scoop: " + cur_val);
-            const p = path.join(home, cur_val.slice(1));
-            // console.log("Expanded ~: " + p);
-            const file = glob.sync(p, "debug").toString();
-            // console.log("Glob for files: " + file);
-
-            if (file) {
-              // console.log("Found some file: " + file);
-              // if there are slashes, reverse them to back slashes
-              constructed_file = file.replace(/\//g, "\\");
-            }
-          } else if (cur_val.startsWith("~")) {
-            constructed_file = path.join(home, cur_val.slice(1));
-            // console.log("Found ~, constructing path: " + constructed_file);
-          } else {
-            constructed_file = cur_val;
-          }
-
-          if (fs.existsSync(constructed_file)) {
-            // console.log("File exists, returning: " + constructed_file);
-            found_nushell_path = constructed_file;
-            break;
-          } else {
-            // console.log("File not found: " + constructed_file);
-          }
-        }
+        const found_nushell_path = which.sync("nu", {
+          nothrow: true,
+          path: pathsToCheck.join(path.delimiter),
+        });
 
-        if (found_nushell_path.length > 0) {
-          return {
-            options: {
-              name: "Nushell",
-              shellPath: found_nushell_path,
-            },
-          };
-        } else {
-          console.log("Nushell not found, returning undefined");
-          return undefined;
+        if (found_nushell_path == null) {
+          console.log(
+            "Nushell not found in env:PATH or any of the heuristic locations."
+          );
         }
+
+        return {
+          options: {
+            name: "Nushell",
+            shellPath: found_nushell_path || "nu",
```

Comment:
same question here about windows `nu` vs `nu.exe`

Reply:
well, if we run to this branch, `found_nushell_path` is null and the search already failed, this one is just used to show an error message to the user. we can change it to something else if you think it's more appropriate. for example, if we change it to `nushell-executable`, the message looks like this

![image](https://github.com/nushell/vscode-nushell-lang/assets/12248559/bbe8c50b-8e97-4efa-9601-bfa1a69faf03)

originally, we return `undefined` for the terminal profile if we failed to locate a `nu` binary, which cause vscode to show a message like this:

![image](https://github.com/nushell/vscode-nushell-lang/assets/12248559/927317c5-be7a-4cfd-bc36-b8bb87f4448b)

it'll pop an error either way, but I think the former is less confusing to the user to figure out what went wrong. not every user knows how to check the log messages.

Reply:
Seems good. I'd just change it to "We cannot find your nushell executable in your path or pre-defined locations.". Something like that.

Reply:
> I'd just change it to "We cannot find your nushell executable in your path or pre-defined locations.".

only the "name" field in the returned terminal profile will showup in the error message. if we want to display a more detailed message, we can do that too.

Reply:
custom error message, together with a link to the website, is added in 8f4af9c